### ISSUE TITLE
Update Dependencies

### ISSUE THREAD

**Issue by niemyjski** [minio/minio-dotnet#211]
I was looking over the project and I had some feedback and questions.

1. Then new project format supports multiple targets of which you can support net452 and .net core in a single csproj which would simplify the whole solution greatly. See: https://github.com/exceptionless/Exceptionless.Net/blob/master/src/Exceptionless/Exceptionless.csproj#L10 Not only this, but you can greatly simplify how nuget packages are created and simplify your build scripts / process.

2. I'm not sure why the project has dependencies on both JSON and XML as well as windows only implementations like build utilities and rergistry. These do not work cross platform: https://github.com/minio/minio-dotnet/blob/master/Minio.Core/Minio.Core.csproj#L117-L122 Can we get these removed? Also RestSharp.NetCore is a third party implementation. I worked with the RestSharp team and now RestSharp 106.x now supports both netstandard and .net full framework. I'd highly recommend doing this ASAP.
```
    <PackageReference Include="Microsoft.Build.Utilities.Core" Version="15.1.548" />
    <PackageReference Include="Microsoft.Win32.Registry" Version="4.3.0" />
    <PackageReference Include="Newtonsoft.Json" Version="10.0.2" />
    <PackageReference Include="RestSharp.NetCore" Version="105.2.3" />
    <PackageReference Include="System.Reactive.Linq" Version="3.1.1" />
    <PackageReference Include="System.Runtime.Serialization.Xml" Version="4.3.0" />
```

If you'd like I'd be more than happy to send over a pr or help maintain this project. I'm working on  https://github.com/FoundatioFx/Foundatio.Minio

**Comment by poornas** [comment]
@niemyjski , thanks for offering to send over a pr or help maintain this project. Both are welcome :+1:

**Comment by deekoder** [comment]
@niemyjski  we don't currently have the cycles to work on this request. If you are able to please send us a PR and we will much appreciate your contribution to minio.

**Comment by bartnuman** [comment]
In addition to this issue; I had an issue where the Main project (web site) had a reference to System.Xml.Linq (.NET framework 4.5.2, so versio 4.0), where the used System.Xml.Linq package referenced by Minio (a port for mono) is version 3.5 (from .NET Framework 2.0). This is causing issue in combination with WebactivatorEx, which is used by Microsoft.Web.Infrastructure...

**Comment by guidoffm** [comment]
I think it is impossible to run the code on Linux because the code accesses the Windows registry that is not available on Linux. 
I have created a version that does not need access to the Windows registry. 

My version works on Linux when it uses the (unsupported) RestSharp.DotNetCore. But this does not work with DotNet 4.x.

The maker of the original RestSharp package claim that RestSharp should work on Linux. But for me it does not. On Linux with the original RestSharp I only get "Forbidden".

To sum up:
Original RestSharp works with DotNet 4.x on Windows and not with DotnetCore on Linux.
RestSharp.DotNetCore  works with DotnetCore on Linux and does not  with DotNet 4.x on Windows.

**Comment by poornas** [comment]
@guidoffm, current master branch of minio-dotnet sdk works on Linux as well as Windows - though it does depend on the unofficial Restsharp.Netcore. The official version of Restsharp had issues originating in upstream HttpClient implementation of DotnetCore. @niemyjski , do you think this PR should be closed until Restsharp issue is resolved?

**Comment by niemyjski** [comment]
I think we should leave it open, as users will know what's going on. I'll follow up with restsharp as I know this issue has been fixed in .net core 2.1 preview 2 and may be back ported.

**Comment by guidoffm** [comment]
For me the question is: Is there a NuGet package that works both on Net461/Windows and NetCore20/Linux.

**Comment by poornas** [comment]
@guidoffm, .net framework can only be run with Mono on Linux, whereas the .netcore package works without additional dependencies on Linux. This is the reason we have two packages for the minio sdk -  [Minio](https://www.nuget.org/packages/Minio/) works on .Net4.5.2 upwards [on windows or with Mono on Linux] and [Minio.Netcore](https://www.nuget.org/packages/Minio.NetCore/) on Linux/Windows.

**Comment by kannappanr** [comment]
Closing this issue as fixed as the corresponding PR #212 has been merged. @niemyjski  Thanks for the awesome work.

**Comment by niemyjski** [comment]
Thanks!

**Comment by niemyjski** [comment]
Can we please get a new release.
